### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by kun4LGit**
- Added focus on first entry on whatever menu we are on. 
focus added

- Updated requirements.txt.
- Updated help menu with project link (can be made clickable later on)
- Icon not displaying but app is stable not crashing

### CODE REVIEW

**Review by Pietrob5 — COMMENTED** [review]
You read my mind with focus on the first input of each page.

> Icon not displaying but app is stable not crashing

I think it's beacuse your folder has a different name to mine. We have to find a way to solve this, but it's a small issue

**Comment by kun4LGit** [comment]
> You read my mind with focus on the first input of each page.
> 
> > Icon not displaying but app is stable not crashing
> 
> I think it's beacuse your folder has a different name to mine. We have to find a way to solve this, but it's a small issue

It can be, Let me think on some ideas too.
hope you find some solution in that link i sent..

**Comment by Pietrob5** [comment]
> hope you find some solution in that link i sent

I'm looking into it and it's useful. It's basically a GUI version of pyinstaller, the tool I previously used, but it doesn't seems to help with our problem. If you want you can try and be more lucky!

**Comment by Pietrob5** [comment]
@kun4LGit Hi! I was thinking about implementing a server-client architecture to use Password-Manager from more devices. Would you be interested in this project? All security stuff will be handled on the client that will send to the server the hased_password, not the clear one. I'm struggling to find an effective way to authenticate the user giving the permession to only acces his own database, not others. Have you got any ideas?

**Comment by kun4LGit** [comment]
> @kun4LGit Hi! I was thinking about implementing a server-client architecture to use Password-Manager from more devices. Would you be interested in this project? All security stuff will be handled on the client that will send to the server the hased_password, not the clear one. I'm struggling to find an effective way to authenticate the user giving the permession to only acces his own database, not others. Have you got any ideas?

Ahh nice idea. I can try my best to help see what i can contribute.
Have you thought of making an API ? 
https://blog.bytebytego.com/p/ep91-rest-api-authentication-methods

Are you planning to use the same Desktop interface we built 
Or have plans to shift to a new web browser based interface. ?
I can make make an auto scalable web interface using HTML + CSS + JS So that user can log in through the website address from any browser be it PC, Tablets or Phones browser. 
But we will need make an API first at least to test some functions as it will be used to handle the data being sent and received from the server.

**Comment by Pietrob5** [comment]
> Are you planning to use the same Desktop interface we built
Or have plans to shift to a new web browser based interface. ?
I can make make an auto scalable web interface using HTML + CSS + JS So that user can log in through the website address from any browser be it PC, Tablets or Phones browser.

For the moment to test we can just use our GUI, then when everithing works we can update to website.
I've never created any API, so I'm new to this field. Can you give me some guidelines?
I was thinking to implement Client-Server communications using TCP, sockets ecc but probably an API is a way better solution.
Your plan is to use the API for all the communications or just for authentiction?

**Comment by kun4LGit** [comment]
> > Are you planning to use the same Desktop interface we built
> > Or have plans to shift to a new web browser based interface. ?
> > I can make make an auto scalable web interface using HTML + CSS + JS So that user can log in through the website address from any browser be it PC, Tablets or Phones browser.
> 
> For the moment to test we can just use our GUI, then when everithing works we can update to website. I've never created any API, so I'm new to this field. Can you give me some guidelines? I was thinking to implement Client-Server communications using TCP, sockets ecc but probably an API is a way better solution. Your plan is to use the API for all the communications or just for authentiction?

Even i have not created API but have seen some vids on it i think it is very much possible and since we are already using python we can use FLASK or Fast API to make the api for app things might seem hard or slow at start but it will be worth the experience as this how things will usually flow in work environment. 
Another advantage is for example lets say you plan to make only android app then since your api is already made you have to just focus on app part not logic. 
In short we will be making something like our main pm.py but for api logic then use it send return data.

> Your plan is to use the API for all the communications or just for authentiction?

best will be to start with authentication but with goal to grow in full api..

May be we both should take few days / week rest and come back with fresh ideas in mind and stuff in coming time. Rather than rushing :D
What say ??

**Comment by Pietrob5** [comment]
> to make the api for app things might seem hard or slow at start but it will be worth the experience as this how things will usually flow in work environment.

100% agree

> May be we both should take few days / week rest and come back with fresh ideas in mind and stuff in coming time. Rather than rushing :D

Sounds good, I'd like to finish this project for october/november so plenty of time. In the meantime  we can look tutorias for this API creation stuff 👍🏻

**Comment by kun4LGit** [comment]
> Sounds good, I'd like to finish this project for october/november so plenty of time. In the meantime we can look tutorias for this API creation stuff 👍🏻

Yes agree 😃💯👍🏼

**Comment by kun4LGit** [comment]
Just quick update 
I have already been working on view all since past few days trying to add searching promt and some progress too till we wait for whole search to complete.. Progress is ok i guess just stuck in positioning and scroll bar.. 
Dont do any changes to it view all for some time im very close :D

**Comment by Pietrob5** [comment]
> Dont do any changes to it view all for some time im very close :D

actually today I pushed a new version in which I added credit card support. I modified the database and functions #1 and #2 in pm_gui.py and added 2 functions in pm.py. It shouldn'be a problem with what you are tryng to do since I used same style, element ecc as always. If you want try and test credit card support and let me know if it works.

**Comment by kun4LGit** [comment]
> > Dont do any changes to it view all for some time im very close :D
> 
> actually today I pushed a new version in which I added credit card support. I modified the database and functions #1 and #2 in pm_gui.py and added 2 functions in pm.py. It shouldn'be a problem with what you are tryng to do since I used same style, element ecc as always. If you want try and test credit card support and let me know if it works.

I will check new updates it should not conflict but still thought i would tell in advance.
I wanted to make those few small updates before I start work on the webpage for API stuff. As i will again have to do quick revisions of JS and also have to check Python API stuff. Many problems we had in tkinter wont even be there when we shift to webapp.
For now i had to remove the function and place it old manner (2 functions) just to get stuff working at this stage..
I cant get the scrollbar working beside that update is fine if you want i can do a pull request. May be you can solve scroll stuff like last time..

**Comment by Pietrob5** [comment]
> For now i had to remove the function and place it old manner (2 functions) just to get stuff working at this stage..

what do you mean?

> I cant get the scrollbar working beside that update is fine if you want i can do a pull request. 

what will be on the PR? What did you chage? And waht's wrong with "scroll stuff"?

**Comment by kun4LGit** [comment]
> what do you mean?
 
> what will be on the PR? What did you chage? And waht's wrong with "scroll stuff"?

Function is split in 2 parts instead of having everything packed in 1 like function inside function.

I added a Searching window with progress bar when we click submit it completes after the full search is done then window changes to display the results. Also fixed the results to display in center.
I also added scroll bar to it but it seem to not be working i think it might be due to the frame_row being pack may be dont know. Beside that everything is working fine.

**Comment by Pietrob5** [comment]
> I added a Searching window with progress bar when we click submit it completes after the full search is done then window changes to display the results. Also fixed the results to display in center.

That's great

> Function` is split in 2 parts instead of having everything packed in 1 like function inside function.

Can you NOT include this part? I like it more like now, with this current indentation of the functions.

> I also added scroll bar to it but it seem to not be working i think it might be due to the frame_row being pack may be dont know.

Scrollbar was already implemented, wasn't it? It works in new windows like for search or view all.
However progress bar and centering are great addons, you can push those 👍🏻

**Comment by kun4LGit** [comment]
> Can you NOT include this part? I like it more like now, with this current indentation of the functions.
> Scrollbar was already implemented, wasn't it? It works in new windows like for search or view all. However progress bar and centering are great addons, you can push those 👍🏻

I had some problems while adding the progress and window to it i just cant seem to fix it i guess im too fatigue or out of ides for now. I was hoping if you can fix it like last time repack function and scrollbar.

Or will just hold on to push and try one more last try to somehow fit all those 3 stuff with existing function again.
As i said sometimes tkinter is really limited and acts weird. We are already running out of space because we choose pack instead of grid and stuff in app like the new add card button in webapp we could have able to fit in top very easily or just like menu. We will have to expand soon i would prefer making new layout then implementing more new stuff and features in this version.

**Comment by Pietrob5** [comment]
>  i would prefer making new layout 

Do you mean new layout in python GUI or in a web app?
I think priority to make it more usable is to create a server to store the database

**Comment by kun4LGit** [comment]
> > i would prefer making new layout
> 
> Do you mean new layout in python GUI or in a web app? I think priority to make it more usable is to create a server to store the database

New layout / Design just browser based or only desktop version through Electron both use HTML CSS Javascript so if we make webpage based we can port it to electron. 

Im retrying to make stuff work in GUI for Searching Label with Progress bar but sadly with indention way you choose it just dont work thats why i had some sub functions out since before it is just some render order issue i guess. 

Another theory i have is when we submit the terminal prints (which it should not as we are in GUI) 
viewall bug
And i feel that is what halting it from displaying searching + progress-bar with the current function indentation as it do displays but post search result. When it should be displaying Searching + Progress as soon as we hit Submit.

**Comment by Pietrob5** [comment]
Don't worry about tkinter GUI

> desktop version through Electron both use HTML CSS Javascript

what you say here is what's important. Do you think is better Electron or Eel? Have you got any knowledge of them? From what I read online, Eel is more suitable for our case, since we already have backend in python.

**Comment by kun4LGit** [comment]
> what you say here is what's important. Do you think is better Electron or Eel? Have you got any knowledge of them? From what I read online, Eel is more suitable for our case, since we already have backend in python.

I have not used Electron or Eel before so i dont know any limitation about them.
I have tiny bit of experience in HTML CSS and JS have made few websites plus just as a practice project worked with API to fetch data for the webpage so i recommended to go on something which can support web based front end and Javascript to fetch / send update plus logic handling like what happen after we press button where does entry data go after click etc. 

I know that Electron has something known as pythonshell which can be used to send receive data from python.
But if API is being created then it wont matter as any data will be get or post by Javascript so backend can be in any language.
Advantages are we can convert to exe easily if im not wrong, Has lots of documentations and videos if we ever need or get stuck.

In Eel we will directly have Python back-end. But im not sure how much documentation is available and is it only for small app or we can push it something very big.

Are you planning to host API ?
or 
Only desktop version application for now ?

**Comment by Pietrob5** [comment]
For the moment only desktop app. I als tought about a flask webapp, but it'd mode difficult to deploy and make it pubblic. A desktop app just need .exe to work

**Comment by kun4LGit** [comment]
> For the moment only desktop app. I als tought about a flask webapp, but it'd mode difficult to deploy and make it pubblic. A desktop app just need .exe to work

If its only desktop based then as of now the current GUI from tkinter is ok.
But if there are plans to add more menu buttons and their different features then will have re design whole entire gui.

There is also a way to skip API and use python builtin sys module and use the stdin, stdout, stderr methods to transport data.
Give me few days / weeks, I will do some research and testing on API if it can run locally and also on sys module and try in both ways and also to see if we can use electron with it. If there is some success we can make new branch and put it as a new version and keep this active till its made.

**Comment by Pietrob5** [comment]
Yes I agree, only I see Eel more suitable than Electron for this project. My opinion

**Comment by Pietrob5** [comment]
@kun4LGit I added a progress animation in view all and the possibility to delete a Credit Card. Have a look and you're welcome to test it, every feedback is like gold

**Comment by kun4LGit** [comment]
> @kun4LGit I added a progress animation in view all and the possibility to delete a Credit Card. Have a look and you're welcome to test it, every feedback is like gold

wow I like it how it was solved i never really thought of threading and stuff.. 👍🏼

**Comment by Pietrob5** [comment]
Maybe threading is a bit to overpower for this task, but it's the easiest solution it came to mind. Did you test the app. Are there any bugs? Try with some wrong inputs and see how it handles them 👍🏻

**Comment by Pietrob5** [comment]
> Maybe threading is a bit to overpower for this task, but it's the easiest solution it came to mind. Did you test the app. Are there any bugs? Try with some wrong inputs and see how it handles them 👍🏻

I also made some major graphic changes, do you like them?

**Comment by kun4LGit** [comment]
> Maybe threading is a bit to overpower for this task, but it's the easiest solution it came to mind. Did you test the app. Are there any bugs? Try with some wrong inputs and see how it handles them 👍🏻

> I also made some major graphic changes, do you like them?
The app is looking so good now :D

I really dont know what updates they did but many years when i tried bootstrap it was lagging a lot but now this feels so smooth infact more fast that the native theme. 
Im very happy to see the progress the app really came a long long way its looking super fine too.. 🥳 

I bit busy but i will try to check and see if any bugs are there but i really feel its on point..
I really like the password suggestion feature its a must for app like this.

Also im already fixing the menu buttons will post update soon 👍🏼

**Comment by Pietrob5** [comment]
> Im very happy to see the progress the app really came a long long way its looking super fine too.. 🥳

Truth

> Also im already fixing the menu buttons will post update soon 👍🏼

What are you doing?

**Comment by kun4LGit** [comment]
> > Im very happy to see the progress the app really came a long long way its looking super fine too.. 🥳
> 
> Truth
> 
> > Also im already fixing the menu buttons will post update soon 👍🏼
> 
> What are you doing?

Fixed the main menu buttons check the pull request.. 
I dont know the boot strap documentation but if i find a way I will try to fix other buttons too..
